notification/search.go

```diff
@@ -98,7 +102,7 @@ var searchTemplate = template.Must(template.New("search").Funcs(search.Helpers()
 		OR (om.created_at = :cursorCreatedAt AND om.id > :afterID)
 	{{end}}
 		AND om.last_status != 'bundled'
-	ORDER BY om.created_at desc, om.id asc
+	ORDER BY (CASE WHEN om.sent_at IS NULL THEN 1 ELSE 0 END) desc, om.created_at desc, om.id asc
```

Comment:
Hmm, this won't entirely give us what we want -- there are failure scenarios where `sent_at` will always be NULL; we also want to sort by the value of `sent_at` and not _only_ if it's null or not.

What about doing `sent_at` but falling back to `last_status_at`? That way, as messages are retried/updated, they will bounce to the top but, once sent, will keep their place.

Also, any messages with the `PENDING` state should probably stay at the top, since those can be important to consider.


```suggestion
	ORDER BY om.last_status = 'pending' desc, coalesce(om.sent_at, om.last_status_at) desc, om.created_at desc, om.id asc
```

Reply:
That makes sense. Change made and test updated as well.